src/wasm/wasm-validator.cpp

```diff
@@ -2355,6 +2355,11 @@ void FunctionValidator::visitRefFunc(RefFunc* curr) {
   shouldBeTrue(!getFunction() || getModule()->features.hasReferenceTypes(),
                curr,
                "ref.func requires reference-types [--enable-reference-types]");
+  if (!shouldBeTrue(curr->type.isRef() && curr->type.isNonNullable(),
```

Comment:
```suggestion
  if (!shouldBeTrue(curr->type.isNonNullable(),
```
`isNonNullable()` already checks `isRef()`. Also below.